ibm_was/datadog_checks/ibm_was/ibm_was.py

```diff
@@ -96,7 +96,10 @@ def submit_metrics(self, child, prefix, tags):
         metric_name = self.normalize(
             ensure_unicode(child.get('name')), prefix='{}.{}'.format(self.METRIC_PREFIX, prefix), fix_case=True
         )
-        self.metric_type_mapping[child.tag](metric_name, value, tags=tags)
+        if prefix == "jvm":
+            self.gauge(metric_name, value, tags=tags)
+        else:
+            self.metric_type_mapping[child.tag](metric_name, value, tags=tags)
```

Comment:
Seems that this will lead to all "jvm" category of metrics being reported as `gauge` instead of using this mapping: https://github.com/DataDog/integrations-core/blob/master/ibm_was/datadog_checks/ibm_was/ibm_was.py#L22-L29

Reply:
We actually want all jvm metrics to be `gauge` (there are only 5 of them).  However, there are some `jdbc` metrics mapped to `CountStatistic` that we want to keep as `count`.  As I say that though, it made me realized that those (correct) count metrics are listed wrong in the metadata.  I'll make another commit to update that.

Reply:
I'm not sure if:

1) This mapping is not correct. 
```python
        self.metric_type_mapping = {
            'AverageStatistic': self.gauge,
            'BoundedRangeStatistic': self.gauge,
            'CountStatistic': self.monotonic_count,
            'DoubleStatistic': self.rate,
            'RangeStatistic': self.gauge,
            'TimeStatistic': self.gauge,
        }
```

2) The mapping above works for some metrics but not others. If that's the case, we might need to reconsider how we map the metrics types. For example, mapping each metric to a type individually like this https://github.com/DataDog/integrations-core/blob/master/hdfs_datanode/datadog_checks/hdfs_datanode/hdfs_datanode.py#L28-L40

(sorry, used my old github account by accident previously)

Reply:
I looked through the XML file here https://github.com/DataDog/integrations-core/blob/master/ibm_was/tests/fixtures/server.xml and it seems the rest of the count metrics were mapped correctly as far as I can tell.  I updated the metadata.csv to reflect this.

Reply:
So `jvm` metrics are of type `CountStatistic` in the xml (extract below), that is mapped to `monotonic_count`, but we want them as gauge. Is that right ?

I'm trying to better understand because adding a condition (exceptional case) like this one makes the code a bit more challenging to understanding and might lead to more cases being added later if other metrics have a similar issue:
```
        if prefix == "jvm":
            self.gauge(metric_name, value, tags=tags)
        else:
            self.metric_type_mapping[child.tag](metric_name, value, tags=tags)
```

If the `jvm` category is indeed the only case, that could be ok, but I'm concerned if there are more similar cases.

---

```
<Stat name="JVM Runtime">
<BoundedRangeStatistic ID="1" highWaterMark="242240" integral="2.05717966272E11" lastSampleTime="1541641692656" lowWaterMark="51200" lowerBound="51200" mean="1063.7910055074533" name="HeapSize" startTime="1541448310718" unit="KILOBYTE" upperBound="262144" value="242240"/>
<CountStatistic ID="2" count="55449" lastSampleTime="1541641692656" name="FreeMemory" startTime="1541448310718" unit="KILOBYTE"/>
<CountStatistic ID="3" count="186790" lastSampleTime="1541641692656" name="UsedMemory" startTime="1541448310718" unit="KILOBYTE"/>
<CountStatistic ID="4" count="193381" lastSampleTime="1541641692656" name="UpTime" startTime="1541448310718" unit="SECOND"/>
<CountStatistic ID="5" count="0" lastSampleTime="1541641692656" name="ProcessCpuUsage" startTime="1541448310718" unit="N/A"/>
</Stat>
```

Reply:
Well, those are the ones identified. You can review the XML file to track other issues.